### PR TITLE
Reader: try showing both direct follow by url and search results

### PR DESCRIPTION
**Pull request by samouri**
Implements https://github.com/Automattic/wp-calypso/issues/13969

### CODE REVIEW

**Comment by bluefuton** [comment]
The 'no results' message feels odd:

screen shot 2017-05-13 at 10 33 26

Should we suppress it in this case?

**Comment by samouri** [comment]
just pushed up a rebase. I'm not sure the answer to your question @bluefuton.
Users may be interested in knowing that their search had no results.

**Comment by bluefuton** [comment]
It's not a question, just an observation. It confused me a bit on first glance! 

I think it's the combination of offering a follow by URL option, and then saying "there are no site results" immediately underneath.

**Comment by jancavan** [comment]
Shouldn't the results (as I type) `nytimes.`, `nytimes.com`, and `http://nytimes.com` be the same or at least similar?

`nytimes.` results are the ones the make the most sense:
screenshot 2017-05-15 14 01 57

`nytimes.com` and `http://nytimes.com` include a bunch of, what seems to be, unrelated to NY Times.

**Comment by jancavan** [comment]
Similar to what @blowery suggested:

screenshot-2017-05-15-14 13 46

**Comment by samouri** [comment]
What do you think of making it more obvious that even though we couldn't find a site,  one may exist?

e.g.: “Sorry, no sites _that we could find_ match nytime.com.  try to follow nytime.com anyway?”

**Comment by gibrown** [comment]
@jancavan the nytimes.com vs nytimes. search results discrepancy is on my list to make better.

**Comment by samouri** [comment]
this is my preferred option over: https://github.com/Automattic/wp-calypso/pull/13957

**Comment by samouri** [comment]
rebased.

**Comment by jancavan** [comment]
Nice work @samouri. If a site/feed is nonexistent, the error isn't helpful at all. It doesn't tell me what the problem is:
screenshot 2017-05-17 12 51 15

What are the scenarios where this error occurs? This is out of scope for this PR though, but just thought of bringing this up.

**Comment by samouri** [comment]
@jancavan, good point. We should probably dispatch different error notices based on what happened: e.g. network error vs. invalid feed.  Network error can keep a similar message but invalid feed could let the user know that the follow will never work.

**Comment by gibrown** [comment]
I think this is a much better experience already and we should probably merge it in, work from there, and find any more bugs with it. There are significant search alg problems it uncovers, but I don't feel like they are blockers.

**Comment by samouri** [comment]
Just did a bit of a wacky rebase.

Could i get a set of 👀 here?

File: client/reader/following-manage/index.jsx [hunk 1]
```diff
@@ -194,6 +195,17 @@ class FollowingManage extends Component {
...
 					{ isSitesQueryUrl &&
 						<div className="following-manage__url-follow">
+							{ isFollowByUrlWithNoSearchResults &&
+								<span className="following-manage__url-follow-no-search-results-message">
+									{ translate(
+										'Sorry, no sites that we could find match {{italic}}%(site1)s{{/italic}}. ' +
+											'Try to follow {{italic}}%(site2)s{{/italic}} anyway?',
+										{
+											components: { italic: <i /> },
+											args: { site1: sitesQuery, site2: sitesQuery },
```
Comment (blowery) [inline client/reader/following-manage/index.jsx]: might want to truncate these?

Comment (samouri) [inline client/reader/following-manage/index.jsx]: not sure what you mean? I think we need to show the whole url right?

Comment (blowery) [inline client/reader/following-manage/index.jsx]: try following `superpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpantssuperpants.com`

Comment (samouri) [inline client/reader/following-manage/index.jsx]: ah, i see.   I'd rather word-wrap!

**Pull request merged**